bin/secret-service/src/seeded_impl/musig2.rs

```diff
@@ -150,17 +153,29 @@ impl Musig2SignerFirstRound<Server, ServerSecondRound> for ServerFirstRound {
         self.first_round.is_complete()
     }
 
-    async fn receive_pub_nonce(
+    async fn receive_pub_nonces(
         &mut self,
-        pubkey: XOnlyPublicKey,
-        pubnonce: musig2::PubNonce,
-    ) -> <Server as Origin>::Container<Result<(), RoundContributionError>> {
-        let signer_idx = self
-            .ordered_public_keys
-            .iter()
-            .position(|x| x == &pubkey)
-            .ok_or(RoundContributionError::out_of_range(0, 0))?;
-        self.first_round.receive_nonce(signer_idx, pubnonce)
+        nonces: HashMap<XOnlyPublicKey, PubNonce>,
+    ) -> <Server as Origin>::Container<Result<(), HashMap<XOnlyPublicKey, RoundContributionError>>>
+    {
+        let mut errs = HashMap::new();
+        for (pubkey, nonce) in nonces {
+            let signer_idx = match self.ordered_public_keys.iter().position(|x| x == &pubkey) {
+                Some(idx) => idx,
+                None => {
+                    errs.insert(pubkey, RoundContributionError::out_of_range(0, 0));
+                    continue;
+                }
+            };
+            if let Err(e) = self.first_round.receive_nonce(signer_idx, nonce) {
+                errs.insert(pubkey, e.clone());
+            }
+        }
```

Comment:
Fun Fact (irrelevant to review but you may find interesting): this is the [Validation Applicative](https://hackage.haskell.org/package/validation-1.1.3/docs/Data-Validation.html#t:Validation)

Reply:
Damn it accumulates errors!